## 主力出货条件统计分析

#### 获取1000支股票数据

In [1]:
import numpy as np
import pandas as pd
import yfinance as yf

In [2]:
df = pd.read_csv('./data/all_data_SS_2y.csv')
df = df.drop(df.columns[0], axis=1)
df

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[399974 rows x 8 columns]

In [3]:
# 过滤掉code列中以'300'、'688'、‘8’开头的股票
df = df[~df['code'].str.startswith('300')]
df = df[~df['code'].str.startswith('688')]
df = df[~df['code'].str.startswith('8')]
df

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[361606 rows x 8 columns]

In [4]:
df.set_index('code', inplace=True)
df

Date   Open   High    Low  Close  Adj Close    Volume
code                                                                  
603123.SS  2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814
603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265
603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283
603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291
603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935
...               ...    ...    ...    ...    ...        ...       ...
600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021
600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100
600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251
600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300
600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420

[361606 rows x 7 columns]

In [5]:
df['Pct_Chg'] = df.groupby(level=0)['Close'].pct_change()
df['Pct_Chg_5d'] = df.groupby(level=0)['Close'].pct_change(periods=5)
df = df.dropna(subset=['Pct_Chg_5d'])
df

Date   Open   High    Low  Close  Adj Close    Volume  \
code                                                                     
603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74  46469718   
603123.SS  2023-01-11  13.87  14.25  13.44  13.67      13.67  40282557   
603123.SS  2023-01-12  13.40  13.64  12.72  12.77      12.77  49225668   
603123.SS  2023-01-13  12.70  12.89  12.64  12.75      12.75  26223929   
603123.SS  2023-01-16  12.85  12.96  12.75  12.85      12.85  22630720   
...               ...    ...    ...    ...    ...        ...       ...   
600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021   
600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100   
600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251   
600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300   
600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420   

            Pct_Chg  Pct_Chg_5d  
code                             
603123.SS -0.039161   -0.084000  
603123.SS -0.005095   -0.051353  
603123.SS -0.065838   -0.152621  
603123.SS -0.001566   -0.102113  
603123.SS  0.007843   -0.101399  
...             ...         ...  
600108.SS  0.016807   -0.012245  
600108.SS -0.008264   -0.004149  
600108.SS  0.008333    0.016807  
600108.SS -0.004132    0.025532  
600108.SS  0.012448    0.025210  

[357096 rows x 9 columns]

In [6]:
stock_data = df.index.get_level_values('code').unique()
stock_data = pd.DataFrame({
    'code': stock_data,
    'is_crash': 0,  # 初始化is_crash列为0
    'is_low': 0
})
stock_data

code  is_crash  is_low
0    603123.SS         0       0
1    600676.SS         0       0
2    600841.SS         0       0
3    601162.SS         0       0
4    600148.SS         0       0
..         ...       ...     ...
896  600370.SS         0       0
897  600202.SS         0       0
898  600985.SS         0       0
899  603188.SS         0       0
900  600108.SS         0       0

[901 rows x 3 columns]

### 这里设置前期暴跌的条件是5日跌20%

In [7]:
df = df.reset_index()
# 遍历df中的每一行
for index, row in df.iterrows():
    # 检查Pct_Chg_5d是否小于-0.2
    if row['Pct_Chg_5d'] < -0.2:
        # 如果是，更新stock_data中的is_crash为1
        stock_data.loc[stock_data['code'] == row['code'], 'is_crash'] = 1
stock_data

code  is_crash  is_low
0    603123.SS         1       0
1    600676.SS         1       0
2    600841.SS         1       0
3    601162.SS         0       0
4    600148.SS         1       0
..         ...       ...     ...
896  600370.SS         1       0
897  600202.SS         1       0
898  600985.SS         0       0
899  603188.SS         1       0
900  600108.SS         0       0

[901 rows x 3 columns]

In [8]:
count_1 = (stock_data['is_crash'] == 1).sum()
count_0 = (stock_data['is_crash'] == 0).sum()
print(f"Number of 1's: {count_1}")
print(f"Number of 0's: {count_0}")

Number of 1's: 398
Number of 0's: 503


### 有398支股票经历过大跌

## 长期慢跌的判断

#### 我认为，如果在过去90天的交易日内，存在60天及以上的数据符合 MA5 < MA10 < MA20 < MA30 ，则可以认为这只股票处于长期的下跌趋势中

In [9]:
df['MA5'] = df['Close'].rolling(window=5).mean()
df['MA10'] = df['Close'].rolling(window=10).mean()
df['MA20'] = df['Close'].rolling(window=20).mean()
df['MA30'] = df['Close'].rolling(window=30).mean()
df = df.dropna(subset=['MA5','MA10','MA20','MA30'])
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000  
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333  
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333  
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333  
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667  
...          ...       ...         ...     ...     ...      ...        ...  
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000  
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667  
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333  
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000  
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667  

[357067 rows x 14 columns]

In [10]:
df['Condition'] = (df['MA5'] < df['MA10']) & (df['MA10'] < df['MA20']) & (df['MA20'] < df['MA30'])
df['Condition'] = df['Condition'].astype(int)
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition  
29              1  
30              0  
31              0  
32              1  
33              1  
...           ...  
357091          0  
357092          0  
357093          0  
357094          0  
357095          0  

[357067 rows x 15 columns]

In [11]:
df['low'] = df['Condition'].rolling(window=90).mean()
df_low = df[df['low'] > 0.6]
df_low

code        Date  Open  High   Low  Close  Adj Close     Volume  \
261     603123.SS  2024-02-05  7.21  7.35  6.62   6.62       6.62   20754101   
262     603123.SS  2024-02-06  6.68  6.88  6.02   6.64       6.64   26037374   
263     603123.SS  2024-02-07  6.70  7.03  6.57   6.80       6.80   25218581   
264     603123.SS  2024-02-08  6.74  7.47  6.73   7.47       7.47   25657747   
265     603123.SS  2024-02-19  7.61  7.76  7.43   7.63       7.63   23458661   
...           ...         ...   ...   ...   ...    ...        ...        ...   
354960  600816.SS  2024-01-31  2.75  3.06  2.72   2.79       2.79  131512384   
355366  600370.SS  2024-02-19  1.76  1.84  1.74   1.84       1.84   20654276   
355367  600370.SS  2024-02-20  1.83  1.86  1.79   1.84       1.84   11433500   
355368  600370.SS  2024-02-21  1.83  1.94  1.81   1.89       1.89   11377183   
355369  600370.SS  2024-02-22  1.89  1.93  1.86   1.92       1.92    8543033   

         Pct_Chg  Pct_Chg_5d    MA5   MA10    MA20      MA30  Condition  \
261    -0.099320   -0.231127  7.630  8.192  8.8360  9.346667          1   
262     0.003021   -0.210464  7.276  8.009  8.6715  9.220667          1   
263     0.024096   -0.144654  7.046  7.813  8.5235  9.108333          1   
264     0.098529   -0.044757  6.976  7.669  8.3985  9.012667          1   
265     0.021419    0.038095  7.032  7.530  8.2945  8.917000          1   
...          ...         ...    ...    ...     ...       ...        ...   
354960 -0.003571   -0.102894  3.116  2.951  2.8610  2.864000          0   
355366  0.063584   -0.031579  1.756  1.905  2.0770  2.181667          1   
355367  0.000000    0.057471  1.776  1.868  2.0510  2.162333          1   
355368  0.027174    0.050000  1.794  1.844  2.0290  2.143667          1   
355369  0.015873    0.149701  1.844  1.833  2.0090  2.126667          0   

             low  
261     0.611111  
262     0.622222  
263     0.633333  
264     0.644444  
265     0.655556  
...          ...  
354960  0.611111  
355366  0.611111  
355367  0.622222  
355368  0.622222  
355369  0.611111  

[7873 rows x 16 columns]

In [12]:
df_low = df_low.reset_index()
# 遍历df中的每一行
for index, row in df_low.iterrows():
    stock_data.loc[stock_data['code'] == row['code'], 'is_low'] = 1
stock_data

code  is_crash  is_low
0    603123.SS         1       1
1    600676.SS         1       0
2    600841.SS         1       0
3    601162.SS         0       0
4    600148.SS         1       0
..         ...       ...     ...
896  600370.SS         1       1
897  600202.SS         1       0
898  600985.SS         0       0
899  603188.SS         1       0
900  600108.SS         0       0

[901 rows x 3 columns]

In [13]:
count_1_low = (stock_data['is_low'] == 1).sum()
count_0_low = (stock_data['is_low'] == 0).sum()
print(f"Number of 1's: {count_1_low}")
print(f"Number of 0's: {count_0_low}")

Number of 1's: 261
Number of 0's: 640


In [14]:
stock_data_condition = stock_data[(stock_data['is_crash'] == 1) & (stock_data['is_low'] == 1)]
stock_data_condition

code  is_crash  is_low
0    603123.SS         1       1
10   603306.SS         1       1
14   600826.SS         1       1
18   600686.SS         1       1
25   600898.SS         1       1
..         ...       ...     ...
877  600722.SS         1       1
880  600355.SS         1       1
891  603755.SS         1       1
895  600816.SS         1       1
896  600370.SS         1       1

[108 rows x 3 columns]

#### 一共有108支股票符合前期暴跌并且长期慢跌的条件

## 计算最低价格

### 最低价格我认为是在前期经历下跌后在低位的涨停前10天左右的最低价格

那么如何确定第一个涨停板呢，在涨停板出现前大概率是下跌趋势

In [15]:
df_copy = pd.read_csv('./data/all_data_SS_2y.csv')
df_copy = df_copy.drop(df_copy.columns[0], axis=1)
df_copy

Date   Open   High    Low  Close  Adj Close    Volume       code
0       2023-01-03  13.96  15.30  13.78  15.00      15.00  67821814  603123.SS
1       2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265  603123.SS
2       2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283  603123.SS
3       2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291  603123.SS
4       2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935  603123.SS
...            ...    ...    ...    ...    ...        ...       ...        ...
399969  2024-08-26   2.38   2.43   2.37   2.42       2.42  13319021  600108.SS
399970  2024-08-27   2.40   2.43   2.39   2.40       2.40   9161100  600108.SS
399971  2024-08-28   2.39   2.43   2.38   2.42       2.42   9000251  600108.SS
399972  2024-08-29   2.42   2.43   2.39   2.41       2.41  10341300  600108.SS
399973  2024-08-30   2.41   2.46   2.39   2.44       2.44  15112420  600108.SS

[399974 rows x 8 columns]

In [16]:
df_copy = df_copy[df_copy['code'].isin(stock_data_condition['code'])]
df_copy.set_index('code', inplace=True)
df_copy['Pct_Chg'] = df_copy.groupby(level=0)['Close'].pct_change()
df_copy = df_copy.dropna(subset=['Pct_Chg'])
df_copy

D:\anaconda3\envs\strategy\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


Date   Open   High    Low  Close  Adj Close    Volume  \
code                                                                     
603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265   
603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283   
603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291   
603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935   
603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74  46469718   
...               ...    ...    ...    ...    ...        ...       ...   
600370.SS  2024-08-26   1.17   1.20   1.17   1.20       1.20   4592800   
600370.SS  2024-08-27   1.20   1.21   1.17   1.19       1.19   5071201   
600370.SS  2024-08-28   1.19   1.22   1.17   1.20       1.20   8253300   
600370.SS  2024-08-29   1.19   1.22   1.18   1.21       1.21   6894200   
600370.SS  2024-08-30   1.21   1.25   1.20   1.24       1.24   9491350   

            Pct_Chg  
code                 
603123.SS -0.039333  
603123.SS  0.045802  
603123.SS -0.057731  
603123.SS  0.007042  
603123.SS -0.039161  
...             ...  
600370.SS  0.016949  
600370.SS -0.008333  
600370.SS  0.008403  
600370.SS  0.008333  
600370.SS  0.024793  

[42617 rows x 8 columns]

In [17]:
min_close_info = df.loc[df.groupby('code')['Close'].idxmin(), ['code', 'Close', 'Date']]
min_close_info.columns = ['code', 'Close_low','Date']
min_close_info

code  Close_low        Date
38971   600000.SS   6.500000  2024-01-12
197836  600004.SS   8.720000  2024-08-22
84059   600006.SS   5.000000  2024-02-05
180734  600007.SS  16.150000  2023-01-10
109979  600008.SS   2.540000  2024-01-22
...           ...        ...         ...
21887   605588.SS  29.209999  2023-01-11
160319  605589.SS  16.100000  2024-02-05
8641    605598.SS  14.450000  2024-02-06
319859  605599.SS   9.640000  2023-02-16
145458  689009.SS  23.340000  2024-01-31

[901 rows x 3 columns]

In [18]:
df_copy = df_copy.merge(min_close_info, on=['code', 'Date'], how='left')
# 添加 is_lowest 列
df_copy['is_lowest'] = df_copy['Close'].eq(df_copy['Close_low']).astype(int)
df_copy

code        Date   Open   High    Low  Close  Adj Close    Volume  \
0      603123.SS  2023-01-04  14.89  14.95  14.35  14.41      14.41  52774265   
1      603123.SS  2023-01-05  14.70  15.58  14.50  15.07      15.07  88029283   
2      603123.SS  2023-01-06  14.69  14.80  14.14  14.20      14.20  58946291   
3      603123.SS  2023-01-09  14.11  14.85  13.95  14.30      14.30  43870935   
4      603123.SS  2023-01-10  14.31  14.31  13.61  13.74      13.74  46469718   
...          ...         ...    ...    ...    ...    ...        ...       ...   
42612  600370.SS  2024-08-26   1.17   1.20   1.17   1.20       1.20   4592800   
42613  600370.SS  2024-08-27   1.20   1.21   1.17   1.19       1.19   5071201   
42614  600370.SS  2024-08-28   1.19   1.22   1.17   1.20       1.20   8253300   
42615  600370.SS  2024-08-29   1.19   1.22   1.18   1.21       1.21   6894200   
42616  600370.SS  2024-08-30   1.21   1.25   1.20   1.24       1.24   9491350   

        Pct_Chg  Close_low  is_lowest  
0     -0.039333        NaN          0  
1      0.045802        NaN          0  
2     -0.057731        NaN          0  
3      0.007042        NaN          0  
4     -0.039161        NaN          0  
...         ...        ...        ...  
42612  0.016949        NaN          0  
42613 -0.008333        NaN          0  
42614  0.008403        NaN          0  
42615  0.008333        NaN          0  
42616  0.024793        NaN          0  

[42617 rows x 11 columns]

写到这里我感觉由最低价得出第一个涨停板过于困难，于是我打算尝试先找涨停板

那么如何区分低位涨停板和高位涨停板呢？

我认为在均线 MA5 < MA10 < MA20 < MA30出现的涨停板可认为是最低点的涨停板

In [19]:
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  
29              1       NaN  
30              0       NaN  
31              0       NaN  
32              1       NaN  
33              1       NaN  
...           ...       ...  
357091          0  0.244444  
357092          0  0.244444  
357093          0  0.233333  
357094          0  0.222222  
357095          0  0.211111  

[357067 rows x 16 columns]

In [23]:
df['满足条件'] = ((df['Condition'] == 1) & (df['Pct_Chg'] > 0.098) & (df['low'] > 0.5)).astype(int)
df

code        Date   Open   High    Low  Close  Adj Close  \
29      603123.SS  2023-02-27  12.33  12.33  12.08  12.09      12.09   
30      603123.SS  2023-02-28  12.26  12.39  12.14  12.25      12.25   
31      603123.SS  2023-03-01  12.26  12.44  12.25  12.35      12.35   
32      603123.SS  2023-03-02  12.35  12.42  12.24  12.26      12.26   
33      603123.SS  2023-03-03  12.26  12.39  12.14  12.34      12.34   
...           ...         ...    ...    ...    ...    ...        ...   
357091  600108.SS  2024-08-26   2.38   2.43   2.37   2.42       2.42   
357092  600108.SS  2024-08-27   2.40   2.43   2.39   2.40       2.40   
357093  600108.SS  2024-08-28   2.39   2.43   2.38   2.42       2.42   
357094  600108.SS  2024-08-29   2.42   2.43   2.39   2.41       2.41   
357095  600108.SS  2024-08-30   2.41   2.46   2.39   2.44       2.44   

          Volume   Pct_Chg  Pct_Chg_5d     MA5    MA10     MA20       MA30  \
29      10209523 -0.021053   -0.024213  12.340  12.520  12.8400  12.859000   
30      10837972  0.013234   -0.016064  12.300  12.445  12.8135  12.809333   
31      15469223  0.008163    0.002435  12.306  12.376  12.7750  12.765333   
32      10754168 -0.007287   -0.018415  12.260  12.320  12.7330  12.748333   
33      11179318  0.006525   -0.000810  12.258  12.329  12.6835  12.734667   
...          ...       ...         ...     ...     ...      ...        ...   
357091  13319021  0.016807   -0.012245   2.388   2.412   2.4295   2.415000   
357092   9161100 -0.008264   -0.004149   2.386   2.407   2.4295   2.414667   
357093   9000251  0.008333    0.016807   2.394   2.407   2.4275   2.414333   
357094  10341300 -0.004132    0.025532   2.406   2.404   2.4255   2.413000   
357095  15112420  0.012448    0.025210   2.418   2.406   2.4250   2.413667   

        Condition       low  满足条件  
29              1       NaN     0  
30              0       NaN     0  
31              0       NaN     0  
32              1       NaN     0  
33              1       NaN     0  
...           ...       ...   ...  
357091          0  0.244444     0  
357092          0  0.244444     0  
357093          0  0.233333     0  
357094          0  0.222222     0  
357095          0  0.211111     0  

[357067 rows x 17 columns]